ui/src/app/shared/modal/save-as-code/update.as.code.component.ts

```diff
@@ -0,0 +1,139 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, NgZone, ViewChild } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { Store } from '@ngxs/store';
+import { ModalTemplate, SuiActiveModal, SuiModalService, TemplateModalConfig } from '@richardlt/ng2-semantic-ui';
+import { Operation } from 'app/model/operation.model';
+import { Pipeline } from 'app/model/pipeline.model';
+import { Project } from 'app/model/project.model';
+import { PipelineService } from 'app/service/pipeline/pipeline.service';
+import { ApplicationWorkflowService } from 'app/service/services.module';
+import { WorkflowService } from 'app/service/workflow/workflow.service';
+import { AutoUnsubscribe } from 'app/shared/decorator/autoUnsubscribe';
+import { ToastService } from 'app/shared/toast/ToastService';
+import { CDSWebWorker } from 'app/shared/worker/web.worker';
+import { AuthenticationState } from 'app/store/authentication.state';
+import { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+
+@Component({
+    selector: 'app-update-ascode',
+    templateUrl: './update-ascode.html',
+    styleUrls: ['./update-ascode.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+@AutoUnsubscribe()
+export class UpdateAsCodeComponent {
+
+    @Input() project: Project;
+    @Input() appName: string;
+    @Input() name: string;
+    dataToSave: any;
+    dataType: string;
+
+    @ViewChild('updateAsCodeModal', { static: false })
+    public myModalTemplate: ModalTemplate<boolean, boolean, void>;
+    modal: SuiActiveModal<boolean, boolean, void>;
+    modalConfig: TemplateModalConfig<boolean, boolean, void>;
+
+    branches: Array<string>;
+    selectedBranch: string;
+    commitMessage: string;
+    loading: boolean;
+    webworkerSub: Subscription;
+    ope: Operation;
+
+    constructor(private _modalService: SuiModalService, private _awService: ApplicationWorkflowService,
+                private _cd: ChangeDetectorRef, private _toast: ToastService, private _translate: TranslateService,
+                private _workflowService: WorkflowService, private _store: Store, private _pipService: PipelineService) {
+    }
+
+    show(data: any, type: string) {
+        this.loading = false;
+        this.dataToSave = data;
+        this.dataType = type;
+        this.modalConfig = new TemplateModalConfig<boolean, boolean, void>(this.myModalTemplate);
+        this.modal = this._modalService.open(this.modalConfig);
+        this._awService.getVCSInfos(this.project.key, this.appName, '').pipe(first())
```

Comment:
no need for loader ?

Reply:
The loader is watching branch variable so it’s ok